dags/destinations/cm360oci.py

```diff
@@ -154,63 +153,85 @@ def _send_payload(self, payload: Dict[str, Any]) -> None:
 
-    valid_conversions = []
-    invalid_conversions = []
+    valid_conversion_tuples = []
+    invalid_conversion_tuples = []
+    http_error = None
 
     for i, entry in enumerate(input_data):
       conversion = {}
-      for conversion_field in CM_CONVERSION_FIELDS:
+      for conversion_field in (CM_REQUIRED_CONVERSIONS_FIELDS + CM_CONVERSION_FIELDS + CM_MUTUALLY_EXCLUSIVE_CONVERSIONS_FIELDS):
         if conversion_field == "timestamp_micros":
           timestamp_micros = self._parse_timestamp_micros(entry)
           if timestamp_micros:
             conversion[conversion_field] = timestamp_micros
           else:
             conversion[conversion_field] = ""
-        elif conversion_field == "conversion_kind":
-          conversion["kind"] = str(entry.get(conversion_field, ""))
         else:
-          conversion[conversion_field] = str(entry.get(conversion_field, ""))
-      
-      if self.validate_conversion(conversion):
-        valid_conversions.append(conversion)
+          value = entry.get(conversion_field)
+          if value:
+            conversion[conversion_field] = value
+
+      is_valid, error_message = self._validate_conversion(conversion)
+      if is_valid:
+        valid_conversion_tuples.append((i, conversion))
       else:
-        invalid_conversions.append((i, errors.ErrorNameIDMap.CM_HOOK_ERROR_INVALID_CONVERSION_EVENT))
+        invalid_conversion_tuples.append((i, error_message))
 
-    if valid_conversions:
+    if valid_conversion_tuples:
       payload = {}
       payload["encryptionInfo"] = self.encryption_info
-      payload["conversions"] = valid_conversions
+      payload["conversions"] = [conversion_tuple[1] for conversion_tuple in valid_conversion_tuples]
 
       if not dry_run:
         try:
           self._send_payload(payload)
         except (
-              errors.DataOutConnectorSendUnsuccessfulError,
-          ) as error:
-            send_error = error.error_num
+            errors.DataOutConnectorSendUnsuccessfulError,
+        ) as error:
+          http_error = error.msg
       else:
         print(
-          "Dry-Run: CM conversions event will not be sent."
+            "Dry-Run: CM conversions event will not be sent."
         )
 
+    if http_error:
+      invalid_conversion_tuples += [(conversion_tuple[0], http_error) for conversion_tuple in valid_conversion_tuples]
+      valid_conversion_tuples = []
+
+    print(f"Sent entries: {len(valid_conversion_tuples)}")
+    print(f"Invalid entries: {len(invalid_conversion_tuples)}")
+
     run_result = RunResult(
-        successful_hits=len(valid_conversions),
-        failed_hits=len(invalid_conversions),
-        error_messages=[str(error[1]) for error in invalid_conversions],
+        successful_hits=len(valid_conversion_tuples),
+        failed_hits=len(invalid_conversion_tuples),
+        error_messages=[str(error[1]) for error in invalid_conversion_tuples],
         dry_run=dry_run,
     )
```

Comment:
as far as I can tell there shouldn't be any changes for the FE, right? it looks like the data in RunResult doesn't change - just wanted to double check.

Reply:
Correct, the format of the RunResult is the same, I'm just adding info to the fields in a different fashion - thanks for confirming!